Can't compile godot without regex support

**Godot version:**
3.2.dev.custom_build. 47ab209b5

**OS/device including version:**
Ubuntu 19.04


**Issue description:**
Can't compile godot with this command
```
scons p=x11 module_bmp_enabled=no module_bullet_enabled=no module_csg_enabled=no module_dds_enabled=no module_enet_enabled=no module_etc_enabled=no module_gdnative_enabled=no module_gridmap_enabled=no module_hdr_enabled=no module_mbedtls_enabled=no module_mobile_vr_enabled=no module_opus_enabled=no module_pvr_enabled=no module_recast_enabled=no module_regex_enabled=no module_squish_enabled=no module_tga_enabled=no module_thekla_unwrap_enabled=no module_theora_enabled=no module_tinyexr_enabled=no module_vorbis_enabled=no module_webm_enabled=no module_websocket_enabled=no -j6
```
because this error shown after line `[ 82%] Linking Program        ==> bin/godot.x11.tools.64`
```
/usr/bin/ld: main/tests/libtests.x11.tools.64.a(test_string.x11.tools.64.o): in function `TestString::test_26()':
/home/rafal/Pulpit/godot/main/tests/test_string.cpp:440: undefined reference to `RegEx::RegEx(String const&)'
/usr/bin/ld: /home/rafal/Pulpit/godot/main/tests/test_string.cpp:441: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'
/usr/bin/ld: /home/rafal/Pulpit/godot/main/tests/test_string.cpp:440: undefined reference to `RegEx::~RegEx()'
/usr/bin/ld: /home/rafal/Pulpit/godot/main/tests/test_string.cpp:440: undefined reference to `RegEx::~RegEx()'
/usr/bin/ld: editor/libeditor.x11.tools.64.a(rename_dialog.x11.tools.64.o): in function `RenameDialog::_regex(String const&, String const&, String const&)':
/home/rafal/Pulpit/godot/editor/rename_dialog.cpp:516: undefined reference to `RegEx::RegEx(String const&)'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:518: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:516: undefined reference to `RegEx::~RegEx()'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:516: undefined reference to `RegEx::~RegEx()'
/usr/bin/ld: editor/libeditor.x11.tools.64.a(rename_dialog.x11.tools.64.o): in function `RenameDialog::_postprocess(String const&)':
/home/rafal/Pulpit/godot/editor/rename_dialog.cpp:536: undefined reference to `RegEx::RegEx(String const&)'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:537: undefined reference to `RegEx::search_all(String const&, int, int) const'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:545: undefined reference to `RegExMatch::get_start(Variant const&) const'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:536: undefined reference to `RegEx::~RegEx()'
/usr/bin/ld: /home/rafal/Pulpit/godot/editor/rename_dialog.cpp:536: undefined reference to `RegEx::~RegEx()'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.x11.tools.64] Error 1
scons: building terminated because of errors.
```

--- a/SConstruct
+++ b/SConstruct
@@ -484,6 +484,13 @@ if selected_platform in platform_list:
     if env['minizip']:
         env.Append(CPPDEFINES=['MINIZIP_ENABLED'])
 
+    editor_module_list = ['regex']
+    for x in editor_module_list:
+        if not env['module_' + x + '_enabled']:
+            if env['tools']:
+                print("Build option 'module_" + x + "_enabled=no' cannot be used with 'tools=yes' (editor), only with 'tools=no' (export template).")
+                sys.exit(255)
+
     if not env['verbose']:
         methods.no_verbose(sys, env)
 

--- a/main/tests/test_string.cpp
+++ b/main/tests/test_string.cpp
@@ -432,6 +432,10 @@ bool test_26() {
 
 	OS::get_singleton()->print("\n\nTest 26: RegEx substitution\n");
 
+#ifndef MODULE_REGEX_ENABLED
+	OS::get_singleton()->print("\tRegEx module disabled, can't run test.");
+	return false;
+#else
 	String s = "Double all the vowels.";
 
 	OS::get_singleton()->print("\tString: %ls\n", s.c_str());
@@ -443,6 +447,7 @@ bool test_26() {
 	OS::get_singleton()->print("\tResult: %ls\n", s.c_str());
 
 	return (s == "Doouublee aall thee vooweels.");
+#endif
 }
 
 struct test_27_data {

